src/jdk.incubator.foreign/share/classes/jdk/incubator/foreign/ResourceScope.java

```diff
@@ -55,21 +56,26 @@
  *     (see {@link CLinker#upcallStub(MethodHandle, FunctionDescriptor, ResourceScope)}</li>
  * </ul>
+ * <p>
+ * Sometimes, explicit scopes can be associated with a {@link Cleaner} instance (see {@link #newConfinedScope(Cleaner)} and
+ * {@link #newSharedScope(Cleaner)}). We call these resource scopes <em>managed</em> resource scopes. A managed resource scope
+ * is closed automatically once the scope instance becomes <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
+ * <p>
+ * Managed scopes can be useful to allow for predictable, deterministic resource deallocation, while still prevent accidental native memory leaks.
+ * In case a managed resource scope is closed explicitly, no further action will be taken when the scope becomes unreachable;
+ * that is, cleanup actions (see {@link #addOnClose(Runnable)}) associated with a resource scope, whether managed or not,
+ * are called <em>exactly once</em>.
  *
- * <h2><a id = "implicit-closure">Implicit closure</a></h2>
+ * <h2>Implicit resource scopes</h2>
  *
- * Resource scopes can be associated with a {@link Cleaner} instance (see {@link #newConfinedScope(Cleaner)}) - we call these
- * resource scopes <em>managed</em> resource scopes. A managed resource scope is closed automatically once the scope instance
- * becomes <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
- * <p>
- * Managed resource scopes can still be closed explicitly (see {@link #close()}); this can be useful to allow for predictable,
- * deterministic resource deallocation, while still prevent accidental native memory leaks. In case a managed resource
- * scope is closed explicitly, no further action will be taken when the scope becomes unreachable; that is, cleanup actions
- * (see {@link #addOnClose(Runnable)}) associated with a resource scope, whether managed or not, are called <em>exactly once</em>.
+ * Resource scopes obtained from {@link #newImplicitScope()} cannot be closed explicitly. We call these resource scopes
+ * <em>implicit scopes</em>. Calling {@link #close()} on an implicit resource scope always results in an exception.
+ * Resources associated with implicit scopes are released once the scope instance becomes
+ * <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
  * <p>
- * Some managed resource scopes are implicitly managed (see {@link #newImplicitScope()}, {@link #globalScope()}, and are said to be <em>implicit scopes</em>.
- * An implicit resource scope only features implicit closure, and always throws an {@link UnsupportedOperationException}
- * when the {@link #close()} method is called directly.
+ * An important implicit resource scope is the so called {@link #globalScope() global scope}; the global scope is
+ * an implicit scope that is guaranteed to never become <a href="../../../java/lang/ref/package.html#reachability">unreachable</a>.
+ * As a results, resources associated to the global scope are never cleaned up.
```

Comment:
By "cleaned up" do you mean that no custom clean up actions added to the scope take effect (they are ignored)? A heap segment is associated with the global scope, but it does not imply the heap segment's backing memory (an array) is always kept alive.  We could say that any resource associated with the global scope is "external" and managed independently (by other means), or not at all (such as the native everything segment)

Reply:
So, the global scope is a singleton scope that is always alive. Since it's always alive, it's never closed, which means cleanup actions never get a chance to get called, yes. I'm not sure talking about resources being "external" makes the point clearer? The thing I'm trying to get across is that global scope is good for things that do **not** require cleanup (e.g. are assumed to be alive for the entire duration of the application).